sdk/servicebus/service-bus/src/util/utils.ts

```diff
@@ -485,3 +487,105 @@ export type EntityStatus =
   | "Renaming"
   | "Restoring"
   | "Unknown";
+
+/**
+ * An executor for a Promise (or function that returns a Promise) that obeys both a timeout and an
+ * optional AbortSignal.
+ * @param timeoutMs - The number of milliseconds to allow before throwing an AMQP error with condition ServiceUnavailableError
+ * @param timeoutMessage - The message to place in the .description field for the thrown exception for Timeout.
+ * @param abortSignal - The abortSignal associated with containing operation.
+ * @param abortErrorMsg - The abort error message associated with containing operation.
+ * @param value - The value to be resolved with after a timeout of t milliseconds.
+ * @returns {Promise<T>} - Resolved promise
+ *
+ * @internal
+ * @ignore
+ */
+export function waitForTimeoutAbortOrResolve<T>(args: {
+  actionFn: () => Promise<T>;
+  timeoutMs: number;
+  timeoutMessage: string;
+  abortSignal?: AbortSignalLike;
+  abortMessage?: string;
+}): Promise<T> {
+  if (args.abortSignal && args.abortSignal.aborted) {
+    throw new AbortError(args.abortMessage);
+  }
+
+  let timer: any | undefined = undefined;
+  let clearAbortSignal: (() => void) | undefined = undefined;
+
+  const clearListenersAndTimer = (): void => {
+    clearTimeout(timer);
+
+    if (clearAbortSignal) {
+      clearAbortSignal();
+    }
+  };
+
+  const abortAndTimeoutPromise = new Promise((resolve, reject) => {
+    clearAbortSignal = checkAndRegisterWithAbortSignal(
+      () => {
+        clearTimeout(timer);
+        reject(new AbortError(args.abortMessage));
+      },
+      args.abortMessage,
+      args.abortSignal
+    );
+
+    timer = setTimeout(function clearForTimeout() {
```

Comment:
We generally use the arrow functions for callbacks `setTimeout(() => {...})`

Reply:
Yeah, I used ths variant to make it easier to distinguish this function my tests (ie, I can just check function.name).